src/org/loklak/api/search/SearchServlet.java

```diff
@@ -150,8 +152,9 @@ protected void doGet(final HttpServletRequest request, final HttpServletResponse
                                     DAO.getConfig(SEARCH_MAX_LOCALHOST_COUNT_NAME, 1000) :
                                     DAO.getConfig(SEARCH_MAX_PUBLIC_COUNT_NAME, 100)));
 
-            String filter = post.get("filter", post.get("filter", ""));
-            
+            String filter = post.get("filter", post.get("filter", "")).replaceAll("\\s","");
```

Comment:
Isn't the second `post.get(...)` redundant. If you don't get `filter` parameter in the first one, how are you supposed to get in in the second one?